src/app/server/CommissionManager.h

```diff
@@ -0,0 +1,84 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/server/AppDelegate.h>
+#include <protocols/secure_channel/RendezvousParameters.h>
+
+namespace chip {
+
+enum class ResetFabrics
+{
+    kYes,
+    kNo,
+};
+
+constexpr uint16_t kNoCommissioningTimeout = UINT16_MAX;
+
+enum class CommissioningWindowAdvertisement
+{
+    kBle,
+    kMdns,
+};
+
+class Server;
```

Comment:
Why do forward declaration instead of just including `Server.h`?

Reply:
Since there is mutal inclusion in `Server.h` so we need to foward declare here.